src/cmd_line/node.ts

```diff
@@ -79,21 +103,94 @@ export class LineRange {
+        currentLineNum = vimState.historyTracker.getMark(firstToken.content).position.line;
+        currentColumn = vimState.historyTracker.getMark(firstToken.content).position.character;
+        break;
       default:
         throw new Error('Not Implemented');
     }
+
+    // now handle subsequent tokens, offsetting the current candidate line number
+    for (let tokenIndex = 1; tokenIndex < toks.length; ++tokenIndex) {
+      let currentToken = toks[tokenIndex];
+
+      switch (currentOperation) {
+        case token.TokenType.Plus:
+          switch (currentToken.type) {
+            case token.TokenType.Minus:
+            case token.TokenType.Plus:
+              // undocumented: when there's two operators in a row, vim behaves as though there's a "1" between them
+              currentLineNum += 1;
+              currentColumn = 0;
+              currentOperation = currentToken.type;
+              break;
+            case token.TokenType.Offset:
+              currentLineNum += Number.parseInt(currentToken.content, 10);
+              currentColumn = 0;
+              currentOperation = undefined;
+              break;
+            default:
+              throw Error('Trailing characters');
+          }
+          break;
+        case token.TokenType.Minus:
+          switch (currentToken.type) {
+            case token.TokenType.Minus:
+            case token.TokenType.Plus:
+              // undocumented: when there's two operators in a row, vim behaves as though there's a "1" between them
+              currentLineNum -= 1;
+              currentColumn = 0;
+              currentOperation = currentToken.type;
+              break;
+            case token.TokenType.Offset:
+              currentLineNum -= Number.parseInt(currentToken.content, 10);
+              currentColumn = 0;
+              currentOperation = undefined;
+              break;
+            default:
+              throw Error('Trailing characters');
+          }
+          break;
+        case undefined:
+          switch (currentToken.type) {
+            case token.TokenType.Minus:
+            case token.TokenType.Plus:
+              currentOperation = currentToken.type;
+              break;
+            default:
+              throw Error('Trailing characters');
+          }
+          break;
+      }
+    }
+
+    // undocumented: when there's a trailing operation in the tank without an RHS, vim uses "1"
+    switch (currentOperation) {
+      case token.TokenType.Plus:
+        currentLineNum += 1;
+        currentColumn = 0;
+        break;
+      case token.TokenType.Minus:
+        currentLineNum -= 1;
+        currentColumn = 0;
+        break;
+    }
+
+    // finally, make sure current position is in bounds :)
+    currentLineNum = Math.max(0, currentLineNum);
+    currentLineNum = Math.min(doc.document.lineCount - 1, currentLineNum);
+    return new vscode.Position(currentLineNum, currentColumn);
```

Comment:
can you somehow use isValid in position.ts

Reply:
I just looked and isValid won't tell why it's not valid (<0 vs >MAX).  I looked for something else in there that will push it in-bounds, but didn't find anything.

In reality, this was here before this PR - vim doesn't actually do this (it throws an "E16: Invalid range" instead).  I wasn't sure why this code was here, but I figured whoever put it there knew something I don't about vscode/this extension/etc, and changing this behavior would have been risky and out of the scope of this issue (and already hard-to-review PR), so I left it alone.

LMK if you want me to go the error route instead within this issue or defer this or let sleeping dogs lie.